app/lua/lobject.c

```diff
@@ -68,10 +68,7 @@ int luaO_log2 (unsigned int x) {
   while (x >= 256) { l += 8; x >>= 8; }
   return l + log_2[x];
 #else
- /* Use Normalization Shift Amount Unsigned:  0x1=>31 up to 0xffffffff =>0
-  * See Xtensa Instruction Set Architecture (ISA) Refman  P 462 */
-  asm volatile ("nsau %0, %1;" :"=r"(x) : "r"(x));
-  return 31 - x;
+  return 31 - __builtin_clz(x);
```

Comment:
Same point about use of builtins vs `asm()` as below.

Reply:
Yeah on Xtensa `__builtin_clz()` translates to a `nsau` instruction.